Source/JavaScriptCore/runtime/Completion.cpp

```diff
@@ -319,4 +319,25 @@ HashMap<RefPtr<UniquedStringImpl>, String> retrieveAssertionsFromDynamicImportOp
     return result;
 }
 
+std::optional<ScriptFetchParameters::Type> retrieveTypeAssertion(JSGlobalObject* globalObject, const HashMap<RefPtr<UniquedStringImpl>, String>& assertions)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+
+    if (assertions.isEmpty())
+        return { };
+
+    auto iterator = assertions.find(vm.propertyNames->type.impl());
+    if (iterator == assertions.end())
+        return { };
+
+    String value = iterator->value;
+    if (auto result = ScriptFetchParameters::parseType(value))
+        return result;
+
+    JSC::throwTypeError(globalObject, scope, makeString("Import assertion type\""_s, value, "\" is not valid"_s));
```

Comment:
nit: do we actually need the `JSC::` namespace here?

Reply:
Unnecessary. Removed.